testcluster: clone Settings for each server

RaduBerinde opened the pull request:
#### testserver: unembed Settings, RaftConfig

There is no reason these should be embedded.
Epic: none
Release note: None

#### testcluster: clone Settings for each server

Currently there are many tests where multi-node test clusters share
the same `*Settings` object. This is problematic - on one hand, it can
make some test failures very hard to debug, and on the other hand, we
are missing out on testing more realistic scenarios where nodes in a
cluster don't magically have their settings in sync.

This change adds some test-only code to clone the Settings object.

Fixes #112395
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112456)
<!-- Reviewable:end -->

RaduBerinde committed:
testcluster: clone Settings for each server

Currently there are many tests where multi-node test clusters share
the same `*Settings` object. This is problematic - on one hand, it can
make some test failures very hard to debug, and on the other hand, we
are missing out on testing more realistic scenarios where nodes in a
cluster don't magically have their settings in sync.

This change adds test-only code to clone the Settings object for each
server.

Fixes #112395
Release note: None

RaduBerinde committed:
testserver: unembed Settings, RaftConfig

There is no reason these should be embedded.
Epic: none
Release note: None

RaduBerinde commented:
Friendly ping for whomever has a bit of time to review this. I have a few PRs on top that I hope to merge before I leave on vacation Thursday afternoon. Many thanks!!

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112456#-:-Nh2w5ogE1ntA08YT2mc:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt and @srosenberg)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
